pkg/libartifact/store/store.go

```diff
@@ -200,24 +205,32 @@ func (as ArtifactStore) Add(ctx context.Context, dest string, paths []string, _
 		if err != nil {
 			return nil, err
 		}
-		newArtifactAnnotations := map[string]string{}
-		newArtifactAnnotations[specV1.AnnotationTitle] = filepath.Base(path)
+		// currently we don't allow override of the filename
+		if _, hasTitle := options.Annotations[specV1.AnnotationTitle]; hasTitle {
+			return nil, fmt.Errorf("cannot override filename in artifact %s", options.Annotations[specV1.AnnotationTitle])
+		}
+
+		annots[specV1.AnnotationTitle] = filepath.Base(path)
+
 		newLayer := specV1.Descriptor{
 			MediaType:   detectedType,
 			Digest:      newBlobDigest,
 			Size:        newBlobSize,
-			Annotations: newArtifactAnnotations,
+			Annotations: annots,
 		}
+
 		artifactManifestLayers = append(artifactManifestLayers, newLayer)
 	}
 
 	artifactManifest := specV1.Manifest{
 		Versioned: specs.Versioned{SchemaVersion: 2},
 		MediaType: specV1.MediaTypeImageManifest,
 		// TODO This should probably be configurable once the CLI is capable
-		ArtifactType: "",
-		Config:       specV1.DescriptorEmptyJSON,
-		Layers:       artifactManifestLayers,
+		Config: specV1.DescriptorEmptyJSON,
+		Layers: artifactManifestLayers,
+	}
+	if len(options.ArtifactType) != 0 {
```

Comment:
The if check seems pointless, `artifactManifest.ArtifactType` is already empty here so you can assign always without condition.